### PR TITLE
Voucherify (Actions) and Segment

### PR DESCRIPTION
**Pull request by weronika-kurczyna**
_A summary of your pull request, including the what change you're making and why._

- Changed the structure of events (track, page, screen) - `addCustomEvent`.
   - [x] New requirement for `referral` and `loyalty` objects.
   - [x] We want to place the `source_id` and `email` fields in the customer object. The customer object and source_id are required.
- Changed the structure of the **Identify event**. Previously, fields were passed to the traits object. 
  - [x] Now customer attributes are passed directly, and a default mapping for the address object has been added.
   - [x] the format of some fields was changed to snake_case
- Changed the structure of the **Group event**. 
   - [x] We want to also in this case place the `source_id` and `email` in the customer object. The customer object and source_id are required.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._
- Tests have been changed due to the new changes.
- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Review by joe-ayoub-segment — APPROVED** [review]
LGTM!

**Comment by joe-ayoub-segment** [comment]
Thanks for the PR @weronika-kurczyna , I'll review ASAP.

File: packages/destination-actions/src/destinations/voucherify/addCustomEvent/index.ts [hunk 1]
```diff
@@ -49,18 +94,9 @@ const action: ActionDefinition<Settings, Payload> = {
...
-        'Additional data that will be stored in the [custom event](https://docs.voucherify.io/reference/the-custom-event-object) metadata in Voucherify.',
-      type: 'object',
-      default: {
-        '@path': '$.properties'
-      }
-    },
     type: {
       label: 'Event Type',
-      description: 'Type of the event. It can be track, page or screen.',
+      description: 'Type of the [event](https://segment.com/docs/connections/spec/). It can be Track, Page or Screen.',
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/voucherify/addCustomEvent/index.ts:99]: Hi @weronika-kurczyna , the event type will always be lowercase.

File: packages/destination-actions/src/destinations/voucherify/assignCustomerToGroup/index.ts [hunk 2]
```diff
@@ -8,32 +8,32 @@ const action: ActionDefinition<Settings, Payload> = {
...
+      type: 'object',
+      properties: {
+        source_id: {
+          label: 'Source ID',
+          type: 'string',
+          required: true
+        },
+        email: {
+          label: 'Email',
+          type: 'string'
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/voucherify/assignCustomerToGroup/index.ts:25]: You could add a description here. e.g. 
description: 'The email address for the user'

File: packages/destination-actions/src/destinations/voucherify/assignCustomerToGroup/index.ts [hunk 3]
```diff
@@ -8,32 +8,32 @@ const action: ActionDefinition<Settings, Payload> = {
...
-      label: 'Email Address',
       description:
-        'The email that identifies the [customer](https://docs.voucherify.io/reference/the-customer-object) in Voucherify.',
-      type: 'string',
+        'Object containing information about the [customer](https://docs.voucherify.io/reference/the-customer-object).',
+      type: 'object',
+      properties: {
+        source_id: {
+          label: 'Source ID',
+          type: 'string',
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/voucherify/assignCustomerToGroup/index.ts:20]: You could add a description here so the customer knows what to populate this field with please? e.g. 
description: 'The `source_id` which identifies the [customer](https://docs.voucherify.io/reference/the-customer-object) in Voucherify.'

File: packages/destination-actions/src/destinations/voucherify/addCustomEvent/index.ts [hunk 4]
```diff
@@ -9,31 +9,76 @@ const action: ActionDefinition<Settings, Payload> = {
...
-        '@if': {
-          exists: { '@path': '$.userId' },
-          then: { '@path': '$.userId' },
-          else: { '@path': '$.anonymousId' }
+        source_id: { '@path': '$.userId' },
+        email: {
+          '@if': {
+            exists: { '@path': '$.properties.email' },
+            then: { '@path': '$.properties.email' },
+            else: { '@path': '$.context.traits' }
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/voucherify/addCustomEvent/index.ts:35]: This line should be: 
            else: { '@path': '$.context.traits.email' }

**Review by joe-ayoub-segment — APPROVED** [review]
Looks good. I'll apply the changes I recommended in a separate PR. Merging for deploy.

**Comment by weronika-kurczyna** [comment]
@joe-ayoub-segment thank you very much for making these changes.

**Pull request merged**